packages/waku/src/router/create-pages.ts

```diff
@@ -457,3 +458,277 @@ export function createPages<
+      }
+      const pathSpec = parsePathWithSlug(layout.path);
+      dynamicLayoutPathMap.set(layout.path, [pathSpec, layout.component]);
+    } else {
+      throw new Error('Invalid layout configuration');
+    }
+  };
+
+  const createRoot: CreateRoot = (root) => {
+    if (configured) {
+      throw new Error('createRoot no longer available');
+    }
+    if (rootItem) {
+      throw new Error(`Duplicated root component`);
+    }
+    if (root.render === 'static' || root.render === 'dynamic') {
+      rootItem = root;
+    } else {
+      throw new Error('Invalid root configuration');
+    }
+  };
+
+  let ready: Promise<AllPages | void> | undefined;
+  const configure = async () => {
+    if (!configured && !ready) {
+      ready = fn({ createPage, createLayout, createRoot });
+      await ready;
+      configured = true;
+    }
+    await ready;
+  };
+
+  const definedRouter = new_defineRouter({
+    getPathConfig: async () => {
+      await configure();
+      const paths: {
+        pattern: string;
+        path: PathSpec;
+        components: Record<string, { isStatic: boolean }>;
+        noSsr: boolean;
+      }[] = [];
+      for (const [path, pathSpec] of fixedPathSet) {
+        const noSsr = noSsrSet.has(pathSpec);
+        const isStatic = (() => {
+          for (const [_, [layoutPathSpec]] of dynamicLayoutPathMap) {
+            if (hasPathSpecPrefix(layoutPathSpec, pathSpec)) {
+              return false;
+            }
+          }
+          return true;
+        })();
+
+        paths.push({
+          pattern: path2regexp(parsePathWithSlug(path)),
+          path: pathSpec,
+          components: { 'layout:/': { isStatic } },
+          noSsr,
+        });
+      }
+      for (const [path, [pathSpec]] of dynamicPagePathMap) {
+        const noSsr = noSsrSet.has(pathSpec);
+        paths.push({
+          pattern: path2regexp(parsePathWithSlug(path)),
+          path: pathSpec,
+          components: { 'page:/': { isStatic: false } },
+          noSsr,
+        });
+      }
+      for (const [path, [pathSpec]] of wildcardPagePathMap) {
+        const noSsr = noSsrSet.has(pathSpec);
+        paths.push({
+          pattern: path2regexp(parsePathWithSlug(path)),
+          path: pathSpec,
+          components: { 'page:/': { isStatic: false } },
+          noSsr,
+        });
+      }
+      return paths;
+    },
+    renderRoute: async (path, options) => {
+      await configure();
+
+      const processSkip = <T>(elements: Record<string, T>) => {
+        if (!options.skip) {
+          return elements;
+        }
+        return Object.fromEntries(
+          Object.entries(elements).filter(([k]) => !options.skip!.includes(k)),
+        );
+      };
+
+      // TODO it would be good if routeMap can exist as a lookup for all routes in createPages
+
+      const result: Record<string, ReactNode> = {
+        root: rootItem
+          ? rootItem.component({ children: Children() }) // Is this right?
```

Comment:
We shouldn't call `Children` function.
```suggestion
          ? rootItem.component({ children: createElement(Children) })
```